File: 1760-minimum-limit-of-balls-in-a-bag/1760-minimum-limit-of-balls-in-a-bag.cpp
class Solution {
public:
    bool valid(int mid, vector<int>&nums, int n)
    {
        for(auto i: nums){
            n-= i/mid;
            if(i % mid == 0) n++;
        }
        if(n>=0) return true;
        return false;
    }
    int minimumSize(vector<int>& nums, int maxOperations) {
        
         
        int start = 1;
        int end = *max_element(nums.begin(),nums.end());
        int ans = -1;
        while(start <= end)
        {
            int mid = start + (end - start)/2;
            if(valid(mid, nums, maxOperations))
            {
                ans = mid;
                end = mid - 1;
            }
            else
                start = mid + 1;
            
            
        }
        return ans;
    }
};

File: 1760-minimum-limit-of-balls-in-a-bag/1760-minimum-limit-of-balls-in-a-bag.py
class Solution:
    def isValid(self, mid: int, nums: List[int], n: int) -> bool:
        for i in nums:
            n -= int(i/mid)
            if i % mid == 0:
                n += 1
        if n >= 0:
            return True
        return False



    def minimumSize(self, nums: List[int], maxOperations: int) -> int:
        low = 1
        high = max(nums)
        ans = -1
        while low <= high:
            mid = int(low + (high - low)/2)
            if self.isValid(mid, nums, maxOperations):
                ans = int(mid)
                high = mid - 1
            else:
                low = mid + 1

        return ans
                
        